code/modules/mob/mob_emote.dm

```diff
@@ -166,11 +170,15 @@
 							"<span class='warning'>As you flip your hat falls off!</span>")
 
 	if(prob(5) && ishuman(user))
-		message = "attempts a flip and crashes to the floor!"
-		sleep(0.3 SECONDS)
+		var/turf = get_turf(L)
+		if(isspaceturf(turf))
+			message = "attempts a flip and loses balance!"
+		else
+			message = "attempts a flip and crashes to the floor!"
 		if(istype(L))
+			sleep(0.3 SECONDS)
 			L.Weaken(4 SECONDS)
```

Comment:
We're trying to get rid of sleeps, you can instead use a timer here. This should work the same
```suggestion
			addtimer(CALLBACK(L, TYPE_PROC_REF(/mob/living, Weaken), 4 SECONDS), 0.3 SECONDS)
```